ArgoTests/Tests/DecodedTests.swift

```diff
@@ -116,6 +116,35 @@ class DecodedTests: XCTestCase {
       XCTFail("Unexpected Error Occurred")
     }
   }
+
+  func testDecodedAlternativeWithSuccess_ReturnsSelf() {
+    let user: Decoded<User> = decode(JSONFromFile("user_with_email")!)
+    let user_error: Decoded<User> = decode(JSONFromFile("user_with_bad_type")!)
+    let alternative = user.alternative(user_error)
+
+    XCTAssertEqual(user.value?.id, alternative.value?.id)
+  }
+
+  func testDecodedAlternativeWithError_ReturnsOther() {
+    let user: Decoded<User> = decode(JSONFromFile("user_with_email")!)
+    let user_error: Decoded<User> = decode(JSONFromFile("user_with_bad_type")!)
+    let alternative = user_error.alternative(user)
+
+    XCTAssertEqual(user.value?.id, alternative.value?.id)
```

Comment:
Huh. I would have expected us to have `==` for `Decoded` if the constrained type was equatable. We should probably do that.